pgms/vselbig/vselbig.man
.ds Vi Vista
.ds Vn 2.1
.TH vselbig 1Vi "15 Februar 1994" "\*(Vi Version \*(Vn"
.SH NAME
vselbig \- select largest labelled component
.SH SYNOPSIS
\fBvselbig\fR [\fIinfile\fR] [\fIoutfile\fR]
.SH DESCRIPTION
\fBvselbig\fP
selects the largest labelled component of a labelled image.
.LP
The input image must be a labelled image of type "ubyte" or of type "short".
Typically, it has  been created by some labelling algorithm, such as
"vlabel2d", "vlabel3d", or some segmentation routine, e.g. "visodata".
\fBvselbig\fP then creates a binary output image in which pixels
belonging to the largest component (i.e. the most frequent label value)
have value "1" and all other pixels have value "0".

.SH TYPICAL PROCESSING CHAIN
The following is a typical processing chain:
.LP
visodata -in image.v | vbinarize -min 2 -max 2 | vlabel3d | vselbig -out result.v 
.LP
This chain will segment the input image (visodata). It will then
extract the class labelled "2" (vbinarize) 
and perform a connected component analysis
of this class (vlabel3d). Finally, the largest connected component 
of class "2" is selected using vselbig.

.SH COMMAND LINE OPTIONS
\fBvselbig\fP accepts the following options:
.IP \fB-help\fP 15n
Prints a message describing options.
.IP "\fB-in\fP \fIinfile\fP"
Specifies the Vista data file containing the input images.
.IP "\fB-out\fP \fIoutfile\fP"
Specifies where to write the output images as a Vista data file.
.LP
Input and output files can be specified on the command line or allowed to
default to the standard input and output streams.
.SH "SEE ALSO"
.na
.nh
.BR vlabel2d (1Vi),vlabel3d (1Vi)
.BR VImage (3Vi),
.BR Vista (7Vi)
.hy
.ad
.SH AUTHOR
Gabriele Lohmann <lohmann@cns.mpg.de>


debian/changelog
via (2.0.1svn191-0) stable; urgency=low

  *renamed via-bin to via-utils

 -- Lipsia Group (The official Lipsia developer group) <lipsia@cbs.mpg.de>  Tue, 09 Nov 2010 12:17:00 +0400 

via (2.0.0-0) stable; urgency=low

  *This version comes with the new lipsia-2.0 package
   and comprises some minor bug fixes
  *Fixed type error of VLong (is now long)

 -- Lipsia Group (The official Lipsia developer group) <lipsia@cbs.mpg.de>  Tue, 09 Nov 2010 12:17:00 +0400 

via (1.6.0-2) unstable; urgency=low

  * New internal release

 -- Frank Burkhardt <burk@cbs.mpg.de>  Mon, 20 Jul 2009 11:45:41 +0200

via (1.6.0-1) unstable; urgency=low

  * Initial release (Closes: #521328).
  * New Upstream Version. Renamed 'vcat' to 'vcatobj' to not conflict with
    atfs package in Debian.
  * Update Standards-version to 3.8.2.0 -- no changes necessary.
  * Update watchfile for recent change in upstream naming scheme.
  * Add versioned dependency on debhelper (>= 5) upon recommendation by
    lintian.

 -- Michael Hanke <michael.hanke@gmail.com>  Fri, 19 Jun 2009 13:50:52 +0200

via (1.5.2+svn3529-2) UNRELEASED; urgency=low

  * Add alternative build-dependency on atlas3-base-dev to be able to build
    the package on Debian etch.
  * FTP access to the upstream sources needs passive mode. Fixed debian/watch.

 -- Michael Hanke <michael.hanke@gmail.com>  Tue, 14 Apr 2009 02:46:10 +0200

via (1.5.2+svn3529-1) UNRELEASED; urgency=low

  * New Upstream SVN snapshot that relocates the VIA lib headers to their new
    cannonical location.

 -- Michael Hanke <michael.hanke@gmail.com>  Tue, 31 Mar 2009 22:08:01 +0200

via (1.5.2-1) UNRELEASED; urgency=low

  * Initial Debian packaging. This is a complete rewrite of the original
    Debian packaging effort done by the upstream developers.

 -- Michael Hanke <michael.hanke@gmail.com>  Thu, 26 Mar 2009 19:46:09 +0100


pgms/vselbig/vselbig.c
/****************************************************************
 *
 * Copyright (C) Max Planck Institute 
 * for Human Cognitive and Brain Sciences, Leipzig
 *
 * Author Gabriele Lohmann, 2004, <lipsia@cbs.mpg.de>
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 *
 * $Id: vselbig.c 3177 2008-04-01 14:47:24Z karstenm $
 *
 *****************************************************************/

/*! \brief vselbig - select largest component in a labelled image

\par Description
The input image must be of type "ubyte" or "short".
The output image will be of type "bit" and contain
only those pixels which belong to the most frequent
class.

\par Usage

        <code>vselbig</code>

        \param -in      input image
        \param -out     output image


\par Examples
<br>
\par Known bugs
none.

\file vselbig.c
\author G.Lohmann, MPI-CBS, 2004
*/


/* From the Vista library: */
#include <viaio/Vlib.h>
#include <viaio/mu.h>
#include <viaio/option.h>
#include <via.h>

/* From the standard C libaray: */
#include <stdio.h>
#include <stdlib.h>



int main (int argc, char **argv)
{
  FILE *in_file, *out_file;
  VAttrList list;
  VImage src=NULL, dest=NULL;
  VAttrListPosn posn;
  char prg[50];	
  sprintf(prg,"vselbig V%s", getVersion());
  fprintf (stderr, "%s\n", prg);

  /* Parse command line arguments: */
  VParseFilterCmd (0,NULL, argc, argv,&in_file, &out_file);


  /* Read source image(s): */
  if (! (list = VReadFile (in_file, NULL))) exit (1);
  fclose(in_file);


  for (VFirstAttr (list, & posn); VAttrExists (& posn); VNextAttr (& posn)) {
    if (VGetAttrRepn (& posn) != VImageRepn) continue;
    VGetAttrValue (& posn, NULL, VImageRepn, & src);

    dest = VSelectBig (src,dest);
    if (dest == NULL) VError("destination image is NULL");

    VSetAttrValue (& posn, NULL, VImageRepn, dest);
    VDestroyImage (src);
  }

  /* Write the results to the output file: */
  VHistory(0,NULL,prg,&list,&list);
  if (VWriteFile (out_file, list))
    fprintf (stderr, "%s: done .\n",argv[0]);
  exit(0);
}


